mlir/include/mlir/Dialect/XeGPU/uArch/uArch.h

```diff
@@ -0,0 +1,426 @@
+//===--- uArch.h ---------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+/// \file
+/// Base uArch definition for different architectures.
+///
+//
+//===----------------------------------------------------------------------===//
+#ifndef MLIR_DIALECT_XEGPU_UTILS_UARCH_H
+#define MLIR_DIALECT_XEGPU_UTILS_UARCH_H
+
+#include <any>
+#include <functional>
+#include <iostream>
+#include <map>
+#include <mutex>
+#include <shared_mutex>
+#include <tuple>
+
+#include "mlir/IR/Types.h"
+
+namespace mlir {
+namespace xegpu {
+namespace uArch {
+
+// Data types we need for YAML to uArch translation
+struct Range {
+  int start;
+  int end;
+};
+
+// Tile can be multi-dimensional
+// For example, a 2D tile can be represented as:
+// Tile:
+//   no_of_dims: 2
+//   dim: [2, 2]
+// This represents a 2x2 tile
+struct Tile {
+  uint32_t no_of_dims;
+  std::vector<uint32_t> dims;
+};
+
+// RangeTile represents a range of tiles instead of a single tile
+// RangeTile essentially provides a way represent the supported range of values
+// in each dimension For each dimension, the range of values is represented as a
+// Range For example, a 2D RangeTile can be represented as: RangeTile:
+//   no_of_dims: 2
+//   dims:
+//     - [1, 32]
+//     - [2, 16]
+// This represents a 2x2 RangeTile where the first dimension can have values
+// from 1 to 32 and the second dimension can have values from 2 to 16
+struct RangeTile {
```

Comment:
why not just use DiscreteTile, if this is not used that often.

Reply:
Kept two different ways to specify the range.